src/daft-parquet/src/read.rs

```diff
@@ -192,21 +195,6 @@ async fn read_parquet_single(
         }?;
     }
 
-    let expected_num_columns = if let Some(columns) = requested_columns {
```

Comment:
why are we deleting this verification logic? Amazon asked for this

Reply:
The columns that are passed in may differ from the actual columns read from the parquet file, due to field ID pruning

E.g. if we asked for columns `["foo: 1", "bar: 2"]`, but the parquet file contains `["foo: 1", "bar: 100"]`, then the data that is read will only contain the `["foo: 1"]` column.

Downstream in the code, the remaining columns will be filled with null values. We could instead perform the filling of null values here, if we detect that the columns don't match up.

Reply:
this this check should only run if theres a field_id_mapping? also you could still have the check but have it as 

`requested_columns  <= columns_read` when theres a field_id_mapping

Reply:
Done and reinstated the validations!